rust/core-lib/src/client.rs

```diff
@@ -42,150 +42,176 @@ impl Client {
+        self.0
+            .send_rpc_notification("available_themes", &json!({ "themes": theme_names }))
     }
 
     pub fn theme_changed(&self, name: &str, theme: &ThemeSettings) {
-        self.0.send_rpc_notification("theme_changed",
-                                     &json!({
+        self.0.send_rpc_notification(
+            "theme_changed",
+            &json!({
                                          "name": name,
                                          "theme": theme,
-                                     }));
+                                     }),
+        );
     }
 
     /// Notify the client that a plugin has started.
     pub fn plugin_started(&self, view_id: ViewId, plugin: &str) {
-        self.0.send_rpc_notification("plugin_started",
-                                     &json!({
+        self.0.send_rpc_notification(
+            "plugin_started",
+            &json!({
                                          "view_id": view_id,
                                          "plugin": plugin,
-                                     }));
+                                     }),
+        );
     }
 
     /// Notify the client that a plugin has stopped.
     ///
     /// `code` is not currently used; in the future may be used to
     /// pass an exit code.
     pub fn plugin_stopped(&self, view_id: ViewId, plugin: &str, code: i32) {
-        self.0.send_rpc_notification("plugin_stopped",
-                                     &json!({
+        self.0.send_rpc_notification(
+            "plugin_stopped",
+            &json!({
                                          "view_id": view_id,
                                          "plugin": plugin,
                                          "code": code,
-                                     }));
+                                     }),
+        );
     }
 
     /// Notify the client of the available plugins.
-    pub fn available_plugins(&self, view_id: ViewId,
-                             plugins: &[ClientPluginInfo]) {
-        self.0.send_rpc_notification("available_plugins",
-                                     &json!({
+    pub fn available_plugins(&self, view_id: ViewId, plugins: &[ClientPluginInfo]) {
+        self.0.send_rpc_notification(
+            "available_plugins",
+            &json!({
                                          "view_id": view_id,
-                                         "plugins": plugins }));
+                                         "plugins": plugins }),
+        );
     }
 
-    pub fn update_cmds(&self, view_id: ViewId,
-                       plugin: &str, cmds: &[Command]) {
-        self.0.send_rpc_notification("update_cmds",
-                                     &json!({
+    pub fn update_cmds(&self, view_id: ViewId, plugin: &str, cmds: &[Command]) {
+        self.0.send_rpc_notification(
+            "update_cmds",
+            &json!({
                                          "view_id": view_id,
```

Comment:
it looks like maybe rustfmt is leaving the contents of the macro alone or something? This alignment seems screwy

Reply:
Rust fmt dosen't touch macros so it will have to be done manually